### ISSUE TITLE
Pd installation instructions outdated?

### ISSUE THREAD

**Issue by jarmitage** [cmuartfab/ml-lib#144]
Downloaded the Bela release and trying to follow [this instruction](https://github.com/cmuartfab/ml-lib#installation):

`For Pd, add ml.*.pd_* to your Pd search path and add the "ml" library to your startup flags by opening up the application Preferences and adding "-lib ml" to the Startup Flags.`

- Am I right in thinking that `add ml.*.pd_* to your Pd search path` means just having the folder of `*.pd_darwin` files in the search path? If so this instruction could be simplified a bit.

- Is it necessary to add the startup flag? If I do this I still get the post message `ml can't load library`:

<img width="584" alt="2018-03-18 at 12 02 06" src="https://user-images.githubusercontent.com/2885827/37565653-04f29a64-2aa5-11e8-9f36-a275f043ea99.png">

**Comment by jarmitage** [comment]
Hi @jamiebullock, I tried the new release and added the `-lib ml` flag again. The library appears to load, but I cannot create an `ml` object. Also getting the error `maximum object loading depth 1000 reached`:
<img width="493" alt="2018-03-19 at 10 57 21" src="https://user-images.githubusercontent.com/2885827/37591759-6eca6254-2b64-11e8-897b-e677f495c96c.png">

**Comment by jamiebullock** [comment]
There's no `ml` object as such. The fact it states:

> ml-lib - machine learning library for Max and Pure Data
> version 0.18.0 (c) 2013 Carnegie Mellon University

Shows the library has loaded correctly.

After that you have successfully created an instance of `ml.mlp`. Everything is working!

**Comment by jarmitage** [comment]
Ah great!

Should I worry about that error?

**Comment by jamiebullock** [comment]
No, you're only getting the error because you're trying to load a library `ml` as if it were an object, which isn't supported.

**Comment by jarmitage** [comment]
@jamiebullock is the Linux build also missing a `ml.pd_linux` object?

**Comment by jarmitage** [comment]
@jamiebullock this release titled "Bela Build" doesn't have the ARM32 build: https://github.com/cmuartfab/ml-lib/releases/tag/0.18.0

Any chance you can add that? Can use 0.17.0 for now!

**Comment by jamiebullock** [comment]
Hi @jarmitage it's possible the ml.pd_linux is missing from the Linux release. Do you need it? 

Re: ARM32, yes I can do another generic ARM build, however I'd be interested to know if the Bela build works on other ARM platforms. Might be worth trying!

**Comment by jarmitage** [comment]
We are doing good on Bela with 0.17.0 ARM32 right now!

Are we missing anything substantial not having 0.18.0 + ml.pd_linux?

**Comment by jamiebullock** [comment]
Ah... oops! I made the Bela release for 0.18 without actually attaching the Bela build. I've now done that, so if you go to Releases, there is a Bela-optimised build there.

Interesting that the generic ARM32 build works on Bela. Would be useful to benchmark that against the Bela optimised one.

There are a few other fixes in 0.18, but if you have things working, there's no massive advantage to upgrading.

ml.pd_linux doesn't add any functionality, it's just a convenience if you want to distribute a project with only one external that contains "everything"

**Comment by jarmitage** [comment]
Awesome thank you. Will compare over the weekend and let you know!
